src/client/fontengine.cpp

```diff
@@ -280,15 +282,19 @@ gui::IGUIFont *FontEngine::initFont(const FontSpec &spec)
 	if (spec.italic)
 		setting_suffix.append("_italic");
 
-	u32 size = std::floor(RenderingEngine::getDisplayDensity() *
-			g_settings->getFloat("gui_scaling") * spec.size);
+	u32 size = std::max<u32>(spec.size * RenderingEngine::getDisplayDensity() *
+			g_settings->getFloat("gui_scaling"), 1);
 
-	if (size == 0) {
-		errorstream << "FontEngine: attempt to use font size 0" << std::endl;
-		errorstream << "  display density: " << RenderingEngine::getDisplayDensity() << std::endl;
-		abort();
+	// Constrain the font size to a certain multiple, if necessary
+	u16 divisible_by = 1;
+	g_settings->getU16NoEx(setting_prefix + "font_size_divisible_by", divisible_by);
+	if (divisible_by != 1) {
```

Comment:
```suggestion
	if (divisible_by > 1) {
```
to avoid division by zero